packages/swing-store-lmdb/src/sqlStreamStore.js

```diff
@@ -0,0 +1,148 @@
+// @ts-check
+import sqlite3ambient from 'better-sqlite3';
+import { assert, details as X, q } from '@agoric/assert';
+
+const STREAM_START = { itemCount: 0 };
+
+/**
+ * @param { unknown } streamName
+ * @returns { asserts streamName is string }
+ */
+function insistStreamName(streamName) {
+  assert.typeof(streamName, 'string');
+  assert(streamName.match(/^[-\w]+$/), X`invalid stream name ${q(streamName)}`);
+  return undefined;
+}
+
+/**
+ * @param {unknown} position
+ * @returns { asserts position is StreamPosition }
+ * @typedef {{ itemCount: number }} StreamPosition
+ */
+function insistStreamPosition(position) {
+  assert.typeof(position, 'object');
+  assert(position);
+  assert.typeof(position.itemCount, 'number');
+  assert(position.itemCount >= 0);
+  return undefined;
+}
+
+/**
+ * @param {string} dbDir
+ * @param {{ sqlite3?: typeof import('better-sqlite3') }} [io]
+ */
+export function sqlStreamStore(dbDir, io) {
+  const { sqlite3 = sqlite3ambient } = io || {};
+  const filePath = `${dbDir}/streams.db`; // ISSUE: no path.join?
+  const db = sqlite3(
+    filePath,
+    // { verbose: console.log },
+  );
+  db.exec(`
+    create table if not exists streamItem (
+      streamName text,
+      position integer,
+      item text,
+      primary key (streamName, position)
+    )
+  `);
+
+  const streamStatus = new Map();
+
+  /**
+   * @param {string} streamName
+   * @param {StreamPosition} startPosition
+   * @param {StreamPosition} endPosition
+   */
+  function readStream(streamName, startPosition, endPosition) {
+    insistStreamName(streamName);
+    // TODO: enforce exclusive access? DB transactions are isolated.
+    // assert(
+    //   !streamStatus.get(streamName),
```

Comment:
I'll channel @FUDCo , let's restore this safety check, to prevent concurrent reads/writes/other-reads.

Reply:
Yes.  Now that I'm looking at this commented out code I'm a little confused by what's happening when this code is commented out but some of the other interlock code is not.